include/generic_sort.h

```diff
@@ -41,7 +41,8 @@ std::string to_smtlib(SortKind);
 class GenericSort : public AbsSort
 {
  public:
-  GenericSort(SortKind sk); 
+  GenericSort(SortKind sk);
+  GenericSort(SortKind sk, std::string name_var);
```

Comment:
Please add a comment above this constructor, saying that it is intended only for datatype sorts.